### PR TITLE
Reimplementation of formatSize to return a more meaningful value

### LINKED ISSUE #4198: formatSize has incorrect prefix
formatSize (in strutils.nim) implements the following:
```
  let gigabytes = bytes shr 30
  let megabytes = bytes shr 20
  let kilobytes = bytes shr 10
  if gigabytes != 0:
    result = frmt(gigabytes, megabytes, "GB")
  elif megabytes != 0:
    result = frmt(megabytes, kilobytes, "MB")
  elif kilobytes != 0:
    result = frmt(kilobytes, bytes, "KB")
  else:
    result = insertSep($bytes) & "B"
```

I'm not sure why it stops at GB (and not TB, PB etc), but there are a couple of issues here of questionable importance:

* Kilo is written k, not K, so "KB" should be "kB"
* Strictly speaking, these are binary prefixes, so it would be better to use the IEC ones (kibi etc instead of kilo etc): https://en.wikipedia.org/wiki/Binary_prefix#IEC_prefixes, in which case it should be KiB, MiB, GiB, TiB etc
* Absolutely strictly speaking (okay, maybe I'm getting carried away now), there should be a (non-breaking?) space between the number and the unit: 4096 == "4&nbsp;KiB"

### PR DESCRIPTION
**Pull request by abudden**
Fixes #4198

### CODE REVIEW

File: lib/pure/strutils.nim [hunk 1]
```diff
@@ -1404,6 +1404,8 @@ proc formatBiggestFloat*(f: BiggestFloat, format: FloatFormatMode = ffDefault,
   ## after the decimal point for Nim's ``biggestFloat`` type.
   ##
   ## If ``precision == 0``, it tries to format it nicely.
+  ##
+  ## If ``trim == true``, trailing zeros will be removed.
```
Comment (Araq) [inline lib/pure/strutils.nim]: Please introduce ``proc trimZeros(x: var string)`` instead.

File: web/news.txt [hunk 2]
```diff
@@ -63,6 +68,8 @@ Library Additions
   versions are for ``char`` and ``set[char]``.
 - Added ``splitDecimal`` to ``math.nim`` to split a floating point value
   into an integer part and a floating part (in the range -1<x<1).
+- Added ``trim`` argument to ``formatFloat`` in ``strutils.nim`` to trim
+  trailing zeros.
```
Comment (Araq) [inline web/news.txt]: Meh, please let's not do this. Code bloat.

File: lib/pure/strutils.nim (hunk 1 above)
Comment (abudden) [inline lib/pure/strutils.nim]: Fixed in 1b2c5998540bf5f522619cc532fd416c93b51174

File: web/news.txt (hunk 2 above)
Comment (abudden) [inline web/news.txt]: Fixed in 1b2c5998540bf5f522619cc532fd416c93b51174

File: lib/pure/strutils.nim [hunk 3]
```diff
@@ -1456,28 +1456,82 @@ proc formatFloat*(f: float, format: FloatFormatMode = ffDefault,
   ## after the decimal point for Nim's ``float`` type.
   result = formatBiggestFloat(f, format, precision, decimalSep)
 
-proc formatSize*(bytes: BiggestInt, decimalSep = '.'): string =
-  ## Rounds and formats `bytes`. Examples:
+proc trimZeros*(x: string): string {.noSideEffect.} =
```
Comment (Araq) [inline lib/pure/strutils.nim]: proc trimZeros*(x: var string) {.noSideEffect.}

is more efficient and what you're doing is already slow enough. ;-)

Comment (abudden) [inline lib/pure/strutils.nim]: Fixed in 29c8c839728f216447c49aa26158b004055d4f1d

**Pull request merged**